src/components/content/credential/UpdateCredential.tsx

```diff
@@ -0,0 +1,214 @@
+        form.setFieldsValue({ description: createCredential.description });
+        form.setFieldsValue({ type: createCredential.type });
+        form.setFieldsValue({ variables: createCredential.variables });
+        form.setFieldsValue({ timeToLive: createCredential.timeToLive });
+    };
+
+    const submit = (credentialVariables: CredentialVariables) => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const createCredential: CreateCredential = JSON.parse(JSON.stringify(credentialVariables));
+        createCredential.timeToLive = 3600;
+        const userName: string | null = localStorage.getItem(usernameKey);
+        if (userName === null) {
+            return;
+        }
+        createCredential.xpanseUser = userName;
+        void CredentialsManagementService.updateCredential(createCredential)
+            .then(() => {
+                getCredentials();
+                getTipInfo('success', 'Update Credential Success!');
+                onUpdateCancel();
+            })
+            .catch(() => {
+                getTipInfo('error', 'Update Credential Failed!');
+            });
+    };
+
+    const tailLayout = {
+        wrapperCol: { offset: 8, span: 16, margin: 100 },
+    };
+
+    const setVariablesValue = (index: number, e: ChangeEvent<HTMLInputElement>) => {
+        credentialVariableList[index].value = e.target.value;
+        form.setFieldsValue({ variables: credentialVariableList });
+    };
+
+    useEffect(() => {
+        setFormFields();
+        if (currentCsp === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialTypesByCsp(currentCsp).then((resp) => {
+            if (resp.length > 0) {
+                setCredentialTypeList(resp as CredentialVariables.type[]);
+            }
+        });
+        if (currentType === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialCapabilitiesByCsp(currentCsp, currentType).then((resp) => {
+            if (resp.length > 0) {
+                const credentialVariables: CredentialVariables = resp[0] as CredentialVariables;
+                setCredentialVariableList(credentialVariables.variables);
+                form.setFieldsValue({ variables: credentialVariables.variables });
+            }
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [currentCsp, currentType]);
+
+    const columns: ColumnsType<CredentialVariable> = [
+        {
+            title: 'name',
+            dataIndex: 'name',
+        },
+        {
+            title: 'value',
+            dataIndex: 'value',
+            render: (value, record, index) => (
+                <Form.Item name='value'>
+                    {''}
+                    {record.isSensitive ? (
+                        <Input.Password
+                            defaultValue={record.value}
```

Comment:
in case of sensitive, we can set this to empty. Because when the user clicks on 'show value', he will again see only  masked value which is misleading.

Reply:
this is still not fixed.

Reply:
![image](https://github.com/eclipse-xpanse/xpanse-ui/assets/122504203/1b475855-a4f5-406b-ba0c-c36e5f175cac)
@swaroopar  Hi, When you click to modify, this sensitive variable is invisible

Reply:
API returns always only a dummy ****** for all sensitive values. 
so if user thinks the correct value is displayed behind it, then the credential will be corrupted if an update is done. 
So to make it clear to the user, the senitive fields should have values set to blank. 
And also add tip field with text -> "Actual value for sensitive fields are not available"